app/services/punches_service.rb

```diff
@@ -0,0 +1,15 @@
+class PunchesService
+  def self.create(punches, user)
+    deletes = punches.map { |punch| punch['from'] }
+
+    @punches = user.punches
+    created_punches = @punches.transaction do
+      @punches.by_days(deletes).delete_all
+      @punches.where(
+        company: user.company
+      ).create(punches)
+    end
```

Comment:
You could put this block in a separate method for better readability (e.g. `create_punches(punches)`).

Reply:
Thanks, I will create another method for this